Clean up job data not working on MySQL

**Describe the bug**
If you delete a job, the clean up of the JobEngine data throws an error.

**To Reproduce**
1. Deploy Kapua with a MySQL DB instance instead of the H2, and change parameters accordingly 
2. Create a job, with a target and a step.
3. Run it and wait for it to stop
4. Try to delete a job

**Expected behavior**
Success on delete. 

Instead an exception is thrown.

**Screenshots**
_None_

**Version of Kapua**
1.0.0

**Type of deployment**
Kapua running on MySQL

**Main component affected**
Console

**Additional context**
_None_

Exception:

```
2018-07-06 12:19:33,700 [qtp940060004-20] WARN  o.e.k.a.c.m.a.s.u.KapuaExceptionHandler - RPC service non-application error
org.eclipse.kapua.job.engine.jbatch.exception.CleanJobDataException: Error: 2,1014931817208969211
	at org.eclipse.kapua.job.engine.jbatch.JobEngineServiceJbatch.cleanJobData(JobEngineServiceJbatch.java:221)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.inject.internal.DelegatingInvocationHandler.invoke(DelegatingInvocationHandler.java:50)
	at com.sun.proxy.$Proxy54.cleanJobData(Unknown Source)
	at org.eclipse.kapua.service.job.internal.JobServiceImpl.lambda$delete$5(JobServiceImpl.java:194)
	at org.eclipse.kapua.commons.security.KapuaSecurityUtils$1.call(KapuaSecurityUtils.java:77)
	at org.eclipse.kapua.commons.security.KapuaSecurityUtils$1.call(KapuaSecurityUtils.java:73)
	at org.eclipse.kapua.commons.security.KapuaSecurityUtils.doPrivileged(KapuaSecurityUtils.java:110)
	at org.eclipse.kapua.commons.security.KapuaSecurityUtils.doPrivileged(KapuaSecurityUtils.java:73)
	at org.eclipse.kapua.service.job.internal.JobServiceImpl.delete(JobServiceImpl.java:194)
	at org.eclipse.kapua.app.console.module.job.server.GwtJobServiceImpl.delete(GwtJobServiceImpl.java:188)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:569)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:208)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:248)
	at com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:841)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:206)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449)
	at org.eclipse.kapua.app.console.core.filter.KapuaWebFilter.executeChain(KapuaWebFilter.java:43)
	at org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365)
	at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90)
	at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83)
	at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:383)
	at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362)
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1629)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:219)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
	at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:169)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:564)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:317)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:110)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
	at org.eclipse.jetty.util.thread.Invocable.invokePreferred(Invocable.java:128)
	at org.eclipse.jetty.util.thread.Invocable$InvocableExecutor.invoke(Invocable.java:222)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:294)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:199)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:673)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:591)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.eclipse.kapua.job.engine.jbatch.driver.exception.CleanJobDataDriverException: Error: job-Ag-DhXDHEP7d_s,[Ljava.lang.Object;@71516b06
	at org.eclipse.kapua.job.engine.jbatch.driver.JbatchDriver.cleanJobData(JbatchDriver.java:277)
	at org.eclipse.kapua.job.engine.jbatch.JobEngineServiceJbatch.cleanJobData(JobEngineServiceJbatch.java:219)
	... 67 common frames omitted
Caused by: com.ibm.jbatch.container.exception.PersistenceException: java.sql.SQLException: (conn=41958) Table 'stepexecutioninstancedata' is specified twice, both as a target for 'DELETE' and as a separate source for data
	at org.eclipse.kapua.job.engine.jbatch.persistence.KapuaJDBCPersistenceManagerImpl.purgeByName(KapuaJDBCPersistenceManagerImpl.java:79)
	at org.eclipse.kapua.job.engine.jbatch.driver.JbatchDriver.cleanJobData(JbatchDriver.java:275)
	... 68 common frames omitted
Caused by: java.sql.SQLException: (conn=41958) Table 'stepexecutioninstancedata' is specified twice, both as a target for 'DELETE' and as a separate source for data
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.get(ExceptionMapper.java:198)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.getException(ExceptionMapper.java:110)
	at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:228)
	at org.mariadb.jdbc.MariaDbPreparedStatementClient.executeInternal(MariaDbPreparedStatementClient.java:216)
	at org.mariadb.jdbc.MariaDbPreparedStatementClient.execute(MariaDbPreparedStatementClient.java:150)
	at org.mariadb.jdbc.MariaDbPreparedStatementClient.executeUpdate(MariaDbPreparedStatementClient.java:183)
	at org.eclipse.kapua.job.engine.jbatch.persistence.KapuaJDBCPersistenceManagerImpl.purgeByName(KapuaJDBCPersistenceManagerImpl.java:67)
	... 69 common frames omitted
Caused by: java.sql.SQLException: Table 'stepexecutioninstancedata' is specified twice, both as a target for 'DELETE' and as a separate source for data
Query is: DELETE FROM stepexecutioninstancedata WHERE stepexecid IN (SELECT C.stepexecid FROM jobinstancedata A INNER JOIN executioninstancedata B ON A.jobinstanceid = B.jobinstanceid INNER JOIN stepexecutioninstancedata C ON B.jobexecid = C.jobexecid WHERE A.name = ?), parameters ['job-Ag-DhXDHEP7d_s']
	at org.mariadb.jdbc.internal.util.LogQueryTool.exceptionWithQuery(LogQueryTool.java:153)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:254)
	at org.mariadb.jdbc.MariaDbPreparedStatementClient.executeInternal(MariaDbPreparedStatementClient.java:209)
	... 72 common frames omitted
	
```

--- a/job-engine/jbatch/src/main/java/org/eclipse/kapua/job/engine/jbatch/persistence/KapuaJDBCPersistenceManagerImpl.java
+++ b/job-engine/jbatch/src/main/java/org/eclipse/kapua/job/engine/jbatch/persistence/KapuaJDBCPersistenceManagerImpl.java
@@ -29,14 +29,15 @@ public class KapuaJDBCPersistenceManagerImpl extends JDBCPersistenceManagerImpl 
 
     private static final String CLASSNAME = KapuaJDBCPersistenceManagerImpl.class.getName();
 
+    private static final Logger LOG = Logger.getLogger(CLASSNAME);
+
     private static final String JDBC_CONNECTION_URL = JdbcConnectionUrlResolvers.resolveJdbcUrl();
 
     private static final SystemSetting CONFIG = SystemSetting.getInstance();
     private static final String USERNAME = CONFIG.getString(SystemSettingKey.DB_USERNAME);
     private static final String PASSWORD = CONFIG.getString(SystemSettingKey.DB_PASSWORD);
 
-    private final static Logger logger = Logger.getLogger(CLASSNAME);
-
+    @Override
     protected Connection getConnectionToDefaultSchema() throws SQLException {
         return getConnection();
     }
@@ -47,17 +48,19 @@ public class KapuaJDBCPersistenceManagerImpl extends JDBCPersistenceManagerImpl 
     }
 
     public void purgeByName(String jobName) {
-        logger.entering(CLASSNAME, "purgeByName", jobName);
+        LOG.entering(CLASSNAME, "purgeByName", jobName);
         String deleteJobs = "DELETE FROM jobinstancedata WHERE name = ?";
+
         String deleteJobExecutions = "DELETE FROM executioninstancedata "
                 + "WHERE jobexecid IN ("
-                + "SELECT B.jobexecid FROM jobinstancedata A INNER JOIN executioninstancedata B "
+                + "SELECT B.jobexecid FROM jobinstancedata A INNER JOIN (SELECT * FROM executioninstancedata) B "
                 + "ON A.jobinstanceid = B.jobinstanceid "
                 + "WHERE A.name = ?)";
+
         String deleteStepExecutions = "DELETE FROM stepexecutioninstancedata "
                 + "WHERE stepexecid IN ("
                 + "SELECT C.stepexecid FROM jobinstancedata A INNER JOIN executioninstancedata B "
-                + "ON A.jobinstanceid = B.jobinstanceid INNER JOIN stepexecutioninstancedata C "
+                + "ON A.jobinstanceid = B.jobinstanceid INNER JOIN (SELECT * FROM stepexecutioninstancedata) C "
                 + "ON B.jobexecid = C.jobexecid "
                 + "WHERE A.name = ?)";
 
@@ -78,7 +81,8 @@ public class KapuaJDBCPersistenceManagerImpl extends JDBCPersistenceManagerImpl 
         } catch (SQLException e) {
             throw new PersistenceException(e);
         }
-        logger.exiting(CLASSNAME, "purge");
+
+        LOG.exiting(CLASSNAME, "purgeByName");
     }
 
 }

